pennylane/ops/math/prod.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# pylint: disable=too-few-public-methods,function-redefined
+
+import pennylane as qml
+
+
+class Prod(qml.operation.Operator):
+    """Arithmetic operator class representing the product of two operators."""
```

Comment:
Should we be restricting this class to just two? I don't see anything blocking us from making this class support an arbitrary list of operators. That might also make simplification easier down the road.

In that case `Prod(op1, op2) @ op3` could return effectively `Prod(op1, op2, op3)` instead of `Prod(Prod(op1, op2), op3)`